test/core/files-sharding.spec.js

```diff
@@ -79,22 +83,25 @@ describe('files dir', () => {
           sharding: true
         }
       })
-      ipfs.once('start', done)
+      ipfs.once('ready', done)
     })
 
-    after((done) => ipfs.stop(done))
+    after(function (done) {
+      this.timeout(30 * 1000)
+      ipfs.stop(done)
+    })
 
     it('should be able to add dir with sharding', function (done) {
       this.timeout(30 * 1000)
 
       pull(
-        pull.values(files),
+        pull.values(createTestFiles()),
         ipfs.files.createAddPullStream(),
         pull.collect((err, results) => {
           expect(err).to.not.exist()
           const last = results[results.length - 1]
-          expect(last.path).to.be.eql('test-folder')
-          expect(last.hash).to.be.eql('QmY8TxNWtNViN7syd2DHazPqu21qWfSNjzCDe78e4YMsUD')
+          expect(last.path).to.eql('test-folder')
+          expect(last.hash).to.eql('Qmb3JNLq2KcvDTSGT23qNQkMrr4Y4fYMktHh6DtC7YatLa')
```

Comment:
@pgte had to change this hash because now createTestFiles creates a clean array of files, resulting in a different hash. This means that js-ipfs-unixfs-engine mutates the files that are passed to it, this should be considered a bug.